File: typescript/remove_duplicate.ts


const array: number[] = [0,0,1,1,1,2,2,3,3,4]



const removeDuplicates = (nums: number[])=>{
    
    let uniqueArray: number[] = []

    for(let i = 0; i < nums.length; i++){
        
        if(!uniqueArray.includes(nums[i])){
            uniqueArray.push(nums[i])
        }

    }

   

    let k = uniqueArray.length
    for(let i = 0; i < k; i++){
        nums[i] = uniqueArray[i]
    }
    return k
}

removeDuplicates(array)

File: python/remove_duplicate.py
class Solution(object):

    def removeDuplicates(self, nums):
        expectedNums = []
        if isinstance(nums, list):

            for i in nums:
                if i not in expectedNums:
                    expectedNums.append(i)

            
            

            k = len(expectedNums)

            for i in range(k):
                nums[i] = expectedNums[i]

            print(nums)
            return k

        else:
            print('invalid type of parameter')
            return
    



def main():
    sol = Solution()
    sol.removeDuplicates([0,0,1,1,1,2,2,3,3,4])

main()